src/main/java/software/amazon/cloudwatchlogs/emf/model/MetricDefinitionBuilder.java

```diff
@@ -0,0 +1,123 @@
+        this.unit = unit;
+        this.storageResolution = storageResolution;
+        this.values = values;
+    }
+
+    MetricDefinition(
+            String name, Unit unit, StorageResolution storageResolution, List<Double> values) {
+        this.unit = unit;
+        this.storageResolution = storageResolution;
+        this.values = values;
+        this.name = name;
+    }
+
+    MetricDefinition(String name, Unit unit, StorageResolution storageResolution, Double value) {
+        this.unit = unit;
+        this.storageResolution = storageResolution;
+        this.values = Arrays.asList(value);
+        this.name = name;
+    }
+
+    @Override
+    protected Metric getMetricValuesUnderSize(int size) {
+        List<Double> subList = values.subList(0, size > values.size() ? values.size() : size);
+        return new MetricDefinition(name, unit, storageResolution, subList);
+    }
+
+    @Override
+    protected Metric getMetricValuesOverSize(int size) {
+        if (size > values.size()) {
+            return null;
+        }
+        List<Double> subList = values.subList(size, values.size());
+        return new MetricDefinition(name, unit, storageResolution, subList);
+    }
+
+    /**
+     * @return the values of this metric, simplified to a double instead of a list if there is only
+     *     one value
+     */
+    @Override
+    protected Object getFormattedValues() {
+        return values.size() == 1 ? values.get(0) : values;
+    }
+}
+
+/** Builds MetricDefinition */
+public class MetricDefinitionBuilder extends MetricDefinition implements MetricBuilder {
+
+    MetricDefinitionBuilder(Unit unit, StorageResolution storageResolution, List<Double> values) {
+        super(unit, storageResolution, values);
+    }
+
+    protected MetricDefinitionBuilder(
+            String Name, Unit unit, StorageResolution storageResolution, Double value) {
+        this(unit, storageResolution, Arrays.asList(value));
+        setName(name);
+    }
+
+    MetricDefinitionBuilder(Unit unit, StorageResolution storageResolution) {
+        this(unit, storageResolution, new ArrayList<>());
+    }
+
+    MetricDefinitionBuilder() {
+        this(Unit.NONE, StorageResolution.STANDARD, new ArrayList<>());
+    }
+
+    MetricDefinitionBuilder(double value) {
+        this(Unit.NONE, StorageResolution.STANDARD, value);
+    }
+
+    MetricDefinitionBuilder(Unit unit, double value) {
+        this(unit, StorageResolution.STANDARD, new ArrayList<>(Arrays.asList(value)));
+    }
+
+    MetricDefinitionBuilder(StorageResolution storageResolution, double value) {
+        this(Unit.NONE, storageResolution, new ArrayList<>(Arrays.asList(value)));
+    }
+
+    MetricDefinitionBuilder(Unit unit, StorageResolution storageResolution, double value) {
+        this(unit, storageResolution, new ArrayList<>(Arrays.asList(value)));
+    }
+
+    /** @return a built version of this metric. */
+    public MetricDefinition build() {
```

Comment:
`MetricDefinition` visibility mismatch. Is it public here and package-private above.